### PR TITLE
Css Color preview to display the colors on the gutter area.

### PR DESCRIPTION
**Pull request by devvaannsh**
Adds a feature to display color previews in the gutter for various file types like CSS, LESS, SCSS, and HTML. Providing a quick visual reference for color values. Updates in real-time as changes are made. It can also display mutliple colors at once like in 'linear-gradient' options. Max colors that can be displayed at single line is 4.

Visual references :- 
Screenshot 2024-11-30 105518
Screenshot 2024-11-30 110350
Screenshot 2024-11-30 110434

### CODE REVIEW

File: src/extensionsIntegrated/CSSColorPreview/main.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            CssColorPreview.removeColorMarks();
+        } else {
+            CssColorPreview.showColorMarks();
+        }
+    }
+
+    // init after appReady
+    AppInit.appReady(function () {
+        PreferencesManager.on("change", PREFERENCES_CSS_COLOR_PREVIEW, preferenceChanged);
+        setTimeout(CssColorPreview.init, 1000);
```
Comment (abose) [inline src/extensionsIntegrated/CSSColorPreview/main.js:233]: Why is this set on a timeout?

**Review by abose — CHANGES_REQUESTED** [review]

File: src/extensionsIntegrated/CSSColorPreview/main.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            const editor = CssColorPreview.getEditor();
+            if (editor) {
+                const cm = editor._codeMirror;
+                cm.clearGutter(gutterName);
+            }
+        }
+    };
+
+    function preferenceChanged() {
+        const value = PreferencesManager.get(PREFERENCES_CSS_COLOR_PREVIEW);
```
[the remark this answers is not in the document]
Comment (devvaannsh) [reply src/extensionsIntegrated/CSSColorPreview/main.js:221]: When the active editor is changed, the previous binding gets removed and a new one gets added to the pane. 
Tested thoroughly. Works as expected.

**Comment by devvaannsh** [comment]
Screenshot 2024-11-30 125838

**Comment by abose** [comment]
LGTM Merging. Thanks for contributiong @devvaannsh

**Pull request merged**